### PR TITLE
fix: correction of bug as per issue #24 description

### LINKED ISSUE #24: File without package declaration throws null pointer exception
File which throws exception:
hibernate-orm/tooling/metamodel-generator/src/test/java/DefaultPackageTest.java

The exception is thrown in the ClassInfo class. in method generateClassName line 138.

### PR DESCRIPTION
**Pull request by MaikHenriqueSP**
fixes #24

- when no package is declared, the full class name of a class will be the class name itself
- when no package is declared, the 'package' field in the report will have the value defined as 'unnamed'

### CODE REVIEW

File: src/test/java/com/github/phillima/test/output/TestJSONOutput.java [hunk 1]
```diff
@@ -11,162 +12,177 @@
...
                 .filter(children -> !children.getChildrens().isEmpty())
-				.collect(Collectors.toList());
+                .collect(Collectors.toList());
+
+        assertEquals(0, childrenOfSchema.size());
+    }
+
+    @Test
+    public void testUnnamedPackage() {
+        var unnamedPackageModel = report.getPackages().stream()
```
Comment (felipemandu) [inline src/test/java/com/github/phillima/test/output/TestJSONOutput.java:177]: usei o auto format do IDE e isso alterou toda a config do projeto. As alterações que eu fiz foram

```
    var unnamedPackageModel = report.getPackages().stream()
        .filter(packageModel -> PackageType.UNNAMED.equals(packageModel.getPackageName()))
        .findFirst()
        .get();

    var unnmedClassModel = unnamedPackageModel.getResults().get(0);
 
    assertEquals("unnamed", unnamedPackageModel.getPackageName());
    assertEquals("NoPackageTest", unnmedClassModel.getFullyQualifiedName());
```

File: src/main/java/com/github/phillima/asniffer/ClassInfo.java [hunk 2]
```diff
@@ -152,15 +156,13 @@ private void defineTypeInClassInfo(TypeDeclaration<?> typeDeclaration) {
...
     }
 
-    // if doest not exist a range for the node, return value -1;
     private Integer getLineStart(Node node) {
         Optional<TokenRange> tokenRange = node.getTokenRange();
         if (tokenRange.isPresent()) {
             return tokenRange.get().toRange()
                     .map(range -> range.begin.line)
-                    .orElse(-1);
+                    .orElseThrow(() -> new RuntimeException("Exists a node without a range"));
```
Comment (phillima) [inline src/main/java/com/github/phillima/asniffer/ClassInfo.java:164]: Change to "Node without a range found"

File: src/test/java/com/github/phillima/test/output/TestJSONOutput.java [hunk 3]
```diff
@@ -11,162 +12,177 @@
...
 
-		var childrenOfSchema = packagesContentReport.get(0).getChildrens().stream()
+        var childrenOfSchema = packagesContentReport.get(0).getChildrens().stream()
                 .filter(children -> CodeElementType.SCHEMA.equals(children.getType()))
                 .filter(children -> !children.getChildrens().isEmpty())
-				.collect(Collectors.toList());
+                .collect(Collectors.toList());
+
+        assertEquals(0, childrenOfSchema.size());
+    }
```
Comment (phillima) [inline src/test/java/com/github/phillima/test/output/TestJSONOutput.java:173]: The "testSchemaChildShouldNotHaveChildren" should be put in another test file. A good place would be the class "TestAnnotationSchema".

File: src/test/java/com/github/phillima/test/output/TestJSONOutput.java [hunk 4]
```diff
@@ -11,162 +12,177 @@
...
+
+    @Test
+    public void testUnnamedPackage() {
+        var unnamedPackageModel = report.getPackages().stream()
+                .filter(packageModel -> PackageType.UNNAMED.equals(packageModel.getPackageName()))
+                .findFirst()
+                .get();
+
+        var unnmedClassModel = unnamedPackageModel.getResults().get(0);
```
Comment (phillima) [inline src/test/java/com/github/phillima/test/output/TestJSONOutput.java:182]: I believe the variable should be just "className". Even tough the package is undefined, the class still has a name.

**Review by phillima — CHANGES_REQUESTED** [review]

**Pull request merged**